src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -263,12 +275,20 @@ void fmi3_xml_parse_free_context(fmi3_xml_parser_context_t *context) {
         jm_vector_free(jm_string)(context->attrMapById);
         context->attrMapById = 0;
     }
```

Comment:
These changes were requires to allow freeing the parser context without everything fully setup/allocated, as is the case when terminalsAndIcons does not exist.